WordPress/gutenberg
# Allow themes to use any styles in the theme.json whether or not the block supports it

Implements https://github.com/WordPress/gutenberg/issues/29778#issuecomment-800887310

With this PR, block supports are no longer used to decide whether the styles in the theme.json for a block should be used or not. Block supports are still used to show UI controls in both the block & global styles sidebar (themes can use any style property, users can use use the style properties "supported" by the block).

Changes in this PR:

- Server:
  - Before: block supports were used to decide whether a particular theme.json style was allowed or not.
  - After: block supports are no longer used and **any** theme.json style property that exists can be used in any block (see test instructions).

- Client:
  - Before: block supports were used to decide whether a particular theme.json style was allowed or not + to create UI controls in the block and global sidebars.
  - After: block supports are only used to create UI controls in the block and global sidebars.

- Update theme.json docs accordingly.

## How to test

1. Make sure existing themes work as expected: using TT1-blocks go to the site editor and load the front-end. Verify they work as expected.

2. Make sure themes can use any style property in the theme.json file:

- Create a post / page with a paragraph. Publish.
- Go to the theme.json of the TT1-blocks theme and add this piece of code in the styles section:

```json
"core/paragraph": {
  "typography": {
    "fontFamily": "var(--wp--preset--font-family--cambria-georgia)"
  }
}
```

- Load the front-end and verify the paragraphs get the cambria / georgia family assigned.
- Load the editors and verify the paragraphs get the cambria / georgia familiy assigned. There shouldn't be any font family controls in either the block sidebar or the global sidebar.

Before this change, the font-family for the paragraph will be ignored.
